crates/bevy_winit/src/system.rs

```diff
@@ -0,0 +1,251 @@
+use bevy_ecs::{
+    entity::{Entities, Entity},
+    event::EventWriter,
+    prelude::{Added, Changed, With},
+    system::{Commands, NonSendMut, Query, RemovedComponents, Res},
+};
+use bevy_utils::tracing::{error, info};
+use bevy_window::{
+    Cursor, CursorPosition, PrimaryWindow, Window, WindowClosed, WindowComponents, WindowHandle,
+    WindowMode, WindowPosition, WindowResizeConstraints, WindowResolution, WindowState,
+    WindowTitle,
+};
+use raw_window_handle::HasRawWindowHandle;
+use winit::{
+    dpi::{LogicalSize, PhysicalPosition},
+    event_loop::EventLoopWindowTarget,
+};
+
+use crate::{converters, get_best_videomode, get_fitting_videomode, WinitWindows};
+
+/// System responsible for creating new windows whenever a `Window` component is added
+/// to an entity.
+///
+/// This will default any necessary components if they are not already added.
+pub fn create_window(
+    mut commands: Commands,
+    event_loop: &EventLoopWindowTarget<()>,
+    created_windows: Query<(Entity, WindowComponents), Added<Window>>,
+    mut winit_windows: NonSendMut<WinitWindows>,
+) {
+    for (window_entity, components) in &created_windows {
+        if let Some(_) = winit_windows.get_window(window_entity) {
+            // Just a safe guard
+            continue;
+        }
+
+        info!("Creating a new window: {:?}", window_entity);
+
+        let winit_window = winit_windows.create_window(&event_loop, window_entity, &components);
+
+        commands
+            .entity(window_entity)
+            .insert(WindowHandle::new(winit_window.raw_window_handle()));
+
+        // TODO: Fix this
+        #[cfg(target_arch = "wasm32")]
+        {
+            let channel = world.resource_mut::<web_resize::CanvasParentResizeEventChannel>();
+            if create_window_event.descriptor.fit_canvas_to_parent {
+                let selector = if let Some(selector) = &create_window_event.descriptor.canvas {
+                    selector
+                } else {
+                    web_resize::WINIT_CANVAS_SELECTOR
+                };
+                channel.listen_to_selector(create_window_event.entity, selector);
+            }
+        }
+    }
+}
+
+pub fn despawn_window(
+    mut commands: Commands,
+    entities: &Entities,
+    primary: Option<Res<PrimaryWindow>>,
+    closed: RemovedComponents<Window>,
+    mut close_events: EventWriter<WindowClosed>,
+    mut winit_windows: NonSendMut<WinitWindows>,
+) {
+    for window in closed.iter() {
+        winit_windows.remove_window(window);
+
+        if entities.contains(window) {
+            commands.entity(window).despawn();
+        }
+
+        if let Some(ref primary) = primary {
+            if primary.window == window {
+                commands.remove_resource::<PrimaryWindow>();
+            }
+        }
```

Comment:
If the primary window is being closed but there are more windows currently existing, should the primary window become one of those instead?

Feels like primary window should only stop existing as a resource once there are no more windows left, but I'm not too sure what is the expected / desired behaviour here

Reply:
I still think we should have the primary window as a component marking the 'window spawned by default'

Perhaps we could rename it to 'InitialWindow'?

Reply:
yeah I think I prefer `InitialWindow` over `PrimaryWindow`, "primary" is up to a lot of different considerations that we don't really have power over given that if there are multiple windows in the first place, the user has decided to do that themselves.